### DISCUSSION TITLE
v5 - Asset collection wrong URL?

### DISCUSSION THREAD

**Discussion by gnikolopoulos** [phalcon/cphalcon#16329] [Q&A]
Hi All,

I am upgrading an application from v4.1 to v5.2.1 and I am facing an issue with the asset management. I have code that looks like this in several places so assets are loaded as needed:

```php
$this->assets
->collection('mainCSS')
->addCss('css/styles.css');
```

In v4 this used to work fine and assets would get added with the correct full URL (app is in a sub-directory) like this:

```html
<link rel="stylesheet" type="text/css" href="http://example.com/application/css/styles.css" media="screen">
```
(app is under `http://example.com/application/`)

However, in v5 the output is this:

```html
<link rel="stylesheet" type="text/css" href="http://example.com/css/styles.css" media="screen">
```

I can solve this using a prefix, but I am sure the solution is much simpler and I just can't see it. Can anyone point me in the right direction here?

**Comment by Nazariy** [comment]
I might be wrong but before it was using `getStatic()` method from URL, have you tried to set url prefix? https://docs.phalcon.io/5.0/en/assets#url-prefixes

**Reply by gnikolopoulos** [reply]
The problem with prefixes is that it also gets added to non-local resources. I would think that what worked for v4 would also work for v5... I am sure the issue stems from the fact that the app is not on the document root, another issue that should have been handled in the core by now, but all we have are small hacks in the index.php file...
